ptvsd/_vendored/pydevd/pydevd.py

```diff
@@ -974,11 +974,22 @@ def run(self, file, globals=None, locals=None, is_module=False, set_trace=True):
             module_name = file
             filename = get_fullname(file)
             if filename is None:
-                sys.stderr.write("No module named %s\n" % file)
-                return
+                mod_dir = get_package_dir(module_name)
+                if mod_dir is None:
+                    sys.stderr.write("No module named %s\n" % file)
+                    return
+                else:
+                    filename = get_fullname("%s.__main__" % module_name)
+                    if filename is None:
+                        sys.stderr.write("No module named %s\n" % file)
+                        return
+                    else:
+                        file = filename
             else:
                 file = filename
-            sys.argv[0] = file
+                if filename.endswith('__init__.pyc') or filename.endswith('__init__.py'):
+                    filename = filename.replace('__init__.py', '__main__.py')
```

Comment:
If __main__.pyc exists and not the .py file, then we need to use the pyc counterpart.